src/Lucene.Net/Search/Searcher.cs

```diff
@@ -143,15 +143,56 @@ public virtual Explanation Explain(Query query, int doc, IState state)
 	    /// <seealso cref="Lucene.Net.Search.Similarity.Default">
 	    /// </seealso>
 	    public virtual Similarity Similarity
-	    {
-	        get { return this.similarity; }
-	        set { this.similarity = value; }
-	    }
-
-	    /// <summary> creates a weight for <c>query</c></summary>
-		/// <returns> new weight
-		/// </returns>
-		public /*protected internal*/ virtual Weight CreateWeight(Query query, IState state)
+        {
+            get
+            {
+                if ( usingLightWeightSimilarity)
+                {
+                    return LightWeightSimilarity.Instance;
+                }
+
+                return similarity;
+            }
+            set => similarity = value;
+        }
+
+        /// <summary>
+        /// ThreadStatic flag to indicate if we want to disable idfExplanation
+        /// </summary>
+        [ThreadStatic]
+        protected static bool usingLightWeightSimilarity;
```

Comment:
Need a comment here to explain that we explicitly want it across instances, since we have code that create `new IndexSearcher` and we need to pass that state to new instances.

Reply:
The flag is ThreadStatic mainly because of concurrent usages of the same searcher in different threads. The fact that we don't need an instance to enable it is a welcomed side-effect.

Reply:
`ThreadLocal` vs `ThreadStatic`, I meant, need a comment.